steam-client/src/main/java/com/avenga/steamclient/steam/asyncclient/steamuser/callback/MarketingMessageCallback.java

```diff
@@ -0,0 +1,67 @@
+package com.avenga.steamclient.steam.asyncclient.steamuser.callback;
+
+import com.avenga.steamclient.enums.EMarketingMessageFlags;
+import com.avenga.steamclient.generated.MsgClientMarketingMessageUpdate2;
+import com.avenga.steamclient.model.GlobalID;
+import com.avenga.steamclient.steam.asyncclient.callbackmanager.BaseCallbackMessage;
+import com.avenga.steamclient.util.stream.BinaryReader;
+import com.avenga.steamclient.util.stream.MemoryStream;
+import lombok.Getter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+/**
+ * This callback is fired when the client receives a marketing message update.
+ */
+@Getter
+public class MarketingMessageCallback extends BaseCallbackMessage {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MarketingMessageCallback.class);
+
+    private Date updateTime;
+    private Collection<Message> messages;
+
+    public MarketingMessageCallback(MsgClientMarketingMessageUpdate2 body, byte[] payload) {
+        updateTime = new Date(body.getMarketingMessageUpdateTime() * 1000L);
+
+        List<Message> msgList = new ArrayList<>();
```

Comment:
how big this list might be? Mb it is a good ide to pre-allocate bigger list than the default (size 10)

Reply:
Added pre-allocated size